### DISCUSSION TITLE
Union type in `after:run` plugin no longer works

### DISCUSSION THREAD

**Discussion by katsadim** [cypress-io/cypress#27827] [Questions and Help]
this is mainly a typescript question and it goes like this:

We have the following code in [cypress/cli/types/cypress.d.ts](https://github.com/cypress-io/cypress/blob/c1a8004924af85d889b078b062300d0597e94efe/cli/types/cypress.d.ts#L6069):
```typescript
(action: 'after:run', fn: (results: CypressCommandLine.CypressRunResult | CypressCommandLine.CypressFailedRunResult) => void | Promise<void>): void
```

Lets focus on the type of the `results` argument. This is a union type. And there is no nice way to discriminate the union!

This PR ["breaking: Normalize results from module api, after:run, and after:spec"](https://github.com/cypress-io/cypress/pull/27230/files#top) and more specifically the [removal](https://github.com/cypress-io/cypress/pull/27230/files#diff-1961c785b4523b36516a2cb55556dffe7c39b4dd4aa3e6cf19e413c7c76f4b25L277) of the common field `status` from `CypressRunResult` field in [this](status: 'finished') change seems to be the culprit.

According to the (admitedely deprecated) [TS documentation](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#union-types), we need a common field to discriminate union types. But now this common type is gone!

Before I would do sth like:
```typescript
if (results.status === "failed") {
   // i know that we are talking about `CypressCommandLine.CypressFailedRunResult` so I cast to it without the IDE flaggin this as an error!
} else {
  // it is for sure `CypressCommandLine.CypressRunResult`
}
```

But I am not sure what to do now! Any ideas?

**Comment by JordanWeatherby** [comment]
Did you ever figure this out? this is causing me problems trying to upgrade to cypress v13

**Comment by katsadim** [comment]
Unfortunately i gave up on it.

**Comment by JordanWeatherby** [comment]
I ended up being able to do this to return if it failed or not

```javascript

const resultIsFailure = (
  result: CypressCommandLine.CypressRunResult | CypressCommandLine.CypressFailedRunResult,
) => {
  if ('status' in result) {
    return result.status === 'failed'
  }
  return result.totalFailed > 0
}
```
